deps/openssl/config/README.md

```diff
@@ -0,0 +1,118 @@
+## Upgrading OpenSSL-1.1.0
+
+### Requirements
+- Linux enviroment (Only CentOS7.1 and Ubuntu16 are tested)
+- perl5
+- nasm (http://www.nasm.us/)  The version of 2.11 or higher is needed.
+- GNU as in binutils. The version of 2.26 or higher is needed.
+
+### 0. Check Requirements
+
+```
+$ perl -v
+
+This is perl 5, version 22, subversion 1 (v5.22.1) built for
+x86_64-linux-gnu-thread-multi
+(with 60 registered patches, see perl -V for more detail)
+
+$ as --version
+GNU assembler (GNU Binutils for Ubuntu) 2.26.1
+Copyright (C) 2015 Free Software Foundation, Inc.
+...
+$ nasm -v
+NASM version 2.11.08
+```
+### 1. Obtain and extract new OpenSS sources.
+
+Get a new source from  https://www.openssl.org/source/ and extract
+them into `deps/openssl/openssl`. Then add all files and commit them.
+
+```sh
+$ cd deps/openssl/
+$ rm -rf openssl
+$ tar zxf ~/tmp/openssl-1.1.0h.tar.gz
+$ mv openssl-1.1.0h openssl
+$ git add --all openssl
+$ git commit openssl
+````
+The commit message can be
+
+```
+deps: upgrade openssl sources to 1.1.0h
+
+This replaces all sources of openssl-1.1.0h.tar.gz into
+deps/openssl/openssl
+```
+### 2. Apply a floating patch
+
+Currently, one floating patch is needed to build S390 asm files.
+```
+ commit 094465362758ebf967b33c84d5c96230b46a34b3
+ Author: Shigeki Ohtsu <ohtsu@ohtsu.org>
+ Date:   Wed Mar 7 23:52:52 2018 +0900
+
+     deps: add s390 asm rules for OpenSSL-1.1.0
+
+     This is a floating patch against OpenSSL-1.1.0 to generate asm files
+         with Makefile rules and it is to be submitted to the upstream.
+```
+
+Cherry pick it form the previous commit.
+```sh
+$ git cherry-pick 094465362758ebf967b33c84d5c96230b46a34b3
+```
+### 2. Execute `make` in `deps/openssl/config` directory.
```

Comment:
`2.` -> `3.`

Reply:
`directory.` -> `directory` for consistency with other headings?

Reply:
Fixed.